v2/robotmk/src/session.rs

```diff
@@ -1,67 +1,60 @@
-use super::attempt::Attempt;
 use super::child_process_supervisor::{ChildProcessOutcome, ChildProcessSupervisor, StdioPaths};
+use super::command_spec::CommandSpec;
 use super::config::SessionConfig;
-use super::environment::{Environment, ResultCode};
 use super::termination::TerminationFlag;
+
 use anyhow::Result;
+use camino::{Utf8Path, Utf8PathBuf};
 
-pub enum Session<'a> {
-    Current(CurrentSession<'a>),
+pub enum Session {
+    Current(CurrentSession),
 }
 
-impl Session<'_> {
-    pub fn new<'a>(
-        session_config: &SessionConfig,
-        environment: &'a Environment,
-        termination_flag: &'a TerminationFlag,
-    ) -> Session<'a> {
+impl Session {
+    pub fn new(session_config: &SessionConfig) -> Session {
         match session_config {
-            SessionConfig::Current => Session::<'a>::Current(CurrentSession {
-                environment,
-                termination_flag,
-            }),
+            SessionConfig::Current => Session::Current(CurrentSession {}),
             SessionConfig::SpecificUser(_) => panic!("User sessions not yet implemented!"),
         }
     }
 
-    pub fn run(&self, attempt: &Attempt) -> Result<RunOutcome> {
+    pub fn run(&self, spec: &RunSpec) -> Result<RunOutcome> {
         match self {
-            Self::Current(current_session) => current_session.run(attempt),
+            Self::Current(current_session) => current_session.run(spec),
         }
     }
 }
 
-pub struct CurrentSession<'a> {
-    environment: &'a Environment,
-    termination_flag: &'a TerminationFlag,
+pub struct CurrentSession {}
+
+pub struct RunSpec<'a> {
```

Comment:
I'm open to naming suggestions...

Reply:
My understand is, that this field is unused?

Reply:
Yes, `id` is currently unused, see my comment above.